code/controllers/subsystem/processing/fishing.dm

```diff
@@ -1,4 +1,33 @@
 /// subsystem for the fishing minigame processing.
 PROCESSING_SUBSYSTEM_DEF(fishing)
 	name = "Fishing"
+	flags = SS_BACKGROUND|SS_POST_FIRE_TIMING
 	wait = 0.05 SECONDS // If you raise it to 0.1 SECONDS, you better also modify [datum/fish_movement/move_fish()]
+	///Cached fish properties so we don't have to initalize fish every time
+	var/list/fish_properties
+	///A cache of fish that can be caught by each type of fishing lure
+	var/list/lure_catchables
+
+/datum/controller/subsystem/processing/fishing/Initialize()
+	///init the properties
+	fish_properties = list()
+	for(var/fish_type in subtypesof(/obj/item/fish))
+		var/obj/item/fish/fish = new fish_type(null, FALSE)
+		fish_properties[fish_type] = list()
+		fish_properties[fish_type][FISH_PROPERTIES_FAV_BAIT] = fish.favorite_bait.Copy()
+		fish_properties[fish_type][FISH_PROPERTIES_BAD_BAIT] = fish.disliked_bait.Copy()
+		fish_properties[fish_type][FISH_PROPERTIES_TRAITS] = fish.fish_traits.Copy()
+		qdel(fish)
+
+	///init the list of things lures can catch
+	lure_catchables = list()
+	var/list/fish_types = subtypesof(/obj/item/fish)
+	for(var/lure_type in typesof(/obj/item/fishing_lure))
+		var/obj/item/fishing_lure/lure = new lure_type
+		lure_catchables[lure_type] = list()
+		for(var/obj/item/fish/fish_type as anything in fish_types)
+			if(lure.is_catchable_fish(fish_type, fish_properties[fish_type]))
```

Comment:
You should probably error if this is false right? Is there a usecase for a fish having a favorite bait they can't be caught by?

Reply:
What? No! You're misunderstanding how this works. This is merely merely done to cache the types of fish can be caught by each lure so we don't have to run these checks every time `datum/fish_source/get_modified_fish_table()` is called, but also because examining a fishing lure twice with high enough fishing skill should return a list of fish that can be caught with it.

Reply:
Ah I missed that this said lure and thought it was baits